### DISCUSSION TITLE
How to create example app in react native cli?

### DISCUSSION THREAD

**Discussion by pratul-mindfire** [callstack/react-native-builder-bob#521] [Q&A]
Currently i am creating npm using npx create-react-native-library@latest awesome-library and selected lib type as javascript lib after that is it generating example project in expo so could you please suggest how to create example app in react native cli

**Comment by joaomarcosjova** [comment]
To create an example app in React Native CLI after generating your npm package using `npx create-react-native-library`, you can follow these steps:

1. **Create a New React Native Project:**
   Navigate to the directory where you want to create your example app and run the following command to create a new React Native project:
   ```bash
   npx react-native init ExampleApp
   ```

2. **Navigate to the Example App Directory:**
   After the project is created, navigate into the newly created example app directory:
   ```bash
   cd ExampleApp
   ```

3. **Install Your NPM Package:**
   If your npm package is published on npmjs.com, you can install it directly using npm or yarn. Run the following command to install your npm package:
   ```bash
   npm install your-awesome-library
   ```
   or
   ```bash
   yarn add your-awesome-library
   ```

4. **Import and Use Your Library:**
   Inside your React Native app code, import and use components or functions from your npm package as needed.

5. **Link Native Dependencies (if any):**
   If your npm package has native dependencies, you may need to link them manually. Follow the instructions provided by your library's documentation to link any native dependencies correctly.

6. **Run Your Example App:**
   Once your npm package is installed and imported into your example app, you can run your React Native example app using the following command:
   ```bash
   npx react-native run-android
   ```
   or
   ```bash
   npx react-native run-ios
   ```
   This will start the Metro Bundler and launch your app in an Android or iOS emulator.

7. **Test Your Library Integration:**
   Make sure to thoroughly test your library integration within the example app to ensure that everything works as expected.

By following these steps, you should be able to create an example app in React Native CLI and integrate your npm package for testing and demonstration purposes. Make sure to update the example app code and documentation as needed to showcase the features and capabilities of your library effectively.
